src/UsgsAstroLsSensorModel.cpp

```diff
@@ -704,112 +702,159 @@ csm::ImageCoord UsgsAstroLsSensorModel::groundToImage(
+   //                                       m_detectorNodes.end(),
+   //                                       approxPt.samp);
+   // if (referenceSamp != m_detectorNodes.begin() && referenceSamp == m_detectorNodes.end()) {
+   //   --referenceSamp;
+   // }
+   //
+   // if (referenceSamp != m_detectorNodes.begin()) {
+   //    --referenceSamp;
+   // }
+   // size_t referenceIndex = std::distance(m_detectorNodes.begin(), referenceSamp);
+   //
+   // double a, b, c;
+   // a = m_detectorLineCoeffs[3 * referenceIndex];
+   // b = m_detectorLineCoeffs[3 * referenceIndex + 1];
+   // c = m_detectorLineCoeffs[3 * referenceIndex + 2];
+   //
+   // double d0 = distanceToLine(p0[0], p0[1], a, b, c);
+   // double di = distanceToLine(pi[0], pi[1], a, b, c);
+   // double delta = 0.0;
+   //
+   // csm::ImageCoord lineToCheck;
+   // double lineTime;
+   // std::vector<double> lineDetectorLine;
+   // double detectorDistance;
+//
+   // while (++count < 30) {
+   //
+   //   printf("CHECKING IF CONVERGE @ %f\n", (abs(di) / m_averageDetectorSize));
+   //
+   //   if ((abs(di) / m_averageDetectorSize) < 1.0) {
+   //     printf("TRYING TO CONVERGE @ %f\n", (abs(di) / m_averageDetectorSize));
+   //     lineToCheck = csm::ImageCoord(approxPt.line - 1.0, approxPt.samp);
+   //     lineTime = getImageTime(lineToCheck);
+   //     lineDetectorLine = computeDetectorView(lineTime, groundPt, adj, desiredPrecision);
+   //     detectorDistance = distanceToLine(lineDetectorLine[0], lineDetectorLine[1], a, b, c);
+   //
+   //     std::cout << detectorDistance << " " << di << '\n';
+   //
+   //     if (detectorDistance * di < 0) {
+   //       approxPt.line -= (abs(di) / (abs(di) + abs(detectorDistance)));
+   //       break;
+   //     }
+   //
+   //     lineToCheck = csm::ImageCoord(approxPt.line + 1.0, approxPt.samp);
+   //     lineTime = getImageTime(lineToCheck);
+   //     lineDetectorLine = computeDetectorView(lineTime, groundPt, adj, desiredPrecision);
+   //     detectorDistance = distanceToLine(lineDetectorLine[0], lineDetectorLine[1], a, b, c);
+   //
+   //     std::cout << di << " " << detectorDistance << '\n';
+   //
+   //     if (di * detectorDistance < 0) {
+   //       approxPt.line += (abs(di) / (abs(di) + abs(detectorDistance)));
+   //       break;
+   //     }
+   //   }
+   //
+   //   delta = abs(di) / m_averageDetectorSize;
+   //
+   //   if (d0 * di <= 0) {
+   //     delta = -delta;
+   //   }
+   //
+   //   printf("ORIGINAL LINE: %f\n", approxPt.line);
+   //   printf("DELTA: %f\n", delta);
+   //
+   //   approxPt.line += delta;
+   //   timei = getImageTime(approxPt);
+   //   pi = computeDetectorView(timei, groundPt, adj, desiredPrecision);
+   //   di = distanceToLine(pi[0], pi[1], a, b, c);
+   // }
+   //
+   // if (count >= 30) {
+   //   std::cout << "COULDN'T CONVERGE" << '\n';
+   // }
+
+   std::vector<double> detectorView;
+   double detectorLine = 1;

```

Comment:
Make this like the number of lines in the image. I could see a time when the desired precision is 1 pixel.